src/main/java/com/autilus/hw1006/LinkedListImpl.java

```diff
@@ -0,0 +1,122 @@
+package com.autilus.hw1006;
+
+public class LinkedListImpl<T> implements List<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    private boolean isCorrectIndex(int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("index incorrect " + index);
+        }
+        return true;
+    }
+
+    private Node<T> currentNode(int index) {
```

Comment:
if we want best possible performance we can add if statement:
if(index<size/2).... else...

Reply:
done